compose_rl/registry_builders.py

```diff
@@ -1,19 +1,88 @@
 # Copyright 2024 MosaicML ComposeRL authors
 # SPDX-License-Identifier: Apache-2.0
 
-"""Builds components for all the components in the Compose RL."""
+import copy
+import logging
+from typing import Any, Optional, Union
 
-from compose_rl.registry import KL_CONTROLLER_REGISTRY
+from llmfoundry.utils.registry_utils import construct_from_registry
+from transformers import (
+    PreTrainedTokenizer,
+    PreTrainedTokenizerFast,
+)
 
+from compose_rl import registry
+from compose_rl.ppo.kl_controller import BaseKLController
+from compose_rl.reward_learning import BaseReward
 
-def build_kl_controller(cfg: dict):
-    """Builds the KL Controller from the config."""
-    kl_ctl_type = cfg.get('kl_ctl_type', None)
+__all__ = ['build_kl_controllers', 'build_rewards']
 
-    # Default KL Controller to Fixed Controller
-    if kl_ctl_type is None:
-        kl_ctl_type = 'fixed'
-        cfg['kl_ctl_type'] = kl_ctl_type
-        cfg['init_kl_coef'] = 0.05
+log = logging.getLogger(__name__)
 
-    return KL_CONTROLLER_REGISTRY[kl_ctl_type](cfg)
+Tokenizer = Optional[Union[PreTrainedTokenizer, PreTrainedTokenizerFast]]
+
+
+def build_kl_controllers(
```

Comment:
super nit, i think the builders should be singular (`build_kl_controller`), since when you call it, you are only building one.